programs/leap-util/actions/snapshot.cpp

```diff
@@ -0,0 +1,38 @@
+#include "snapshot.hpp"
+#include <memory>
+
+#include <fc/bitutil.hpp>
+#include <fc/filesystem.hpp>
+#include <fc/io/json.hpp>
+#include <fc/variant.hpp>
+
+#include <boost/exception/diagnostic_information.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/filesystem/path.hpp>
+
+void snapshot_actions::setup(CLI::App& app) {
+   auto* sub = app.add_subcommand("snapshot", "snapshot utility");
+   sub->add_subcommand("to-json", "convert snapshot file to convert to json format");
+
+   // options
+   sub->add_option("--input-file,-i", opt->input_file, "snapshot file to convert to json format, writes to <file>.json if output file not specified (tmp state dir used), and exit.")->required();
```

Comment:
I don't quite follow what this description of the option is trying to tell me. It seems to be saying if I give an input file without an output file, the output file will be basenameofinput.json; but that's in conflict of the `output-file` description which says no output file means output goes to stdout. And I don't know what "tmp state dir" means in this context.

More generally, since the options are attached to `snapshot`, not `to-json`,  the descriptions are probably too narrowly defined solely for the `to-json` use case. There are a number of useful operations on a snapshot file we can add in the future that don't have anything to do with json.